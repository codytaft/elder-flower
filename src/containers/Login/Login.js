import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { setCurrentUser } from '../../actions';
import { getUser } from '../../helpers/fetchCalls';
import './Login.css';

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = e => {
    e.preventDefault();
    const { history } = this.props;
    getUser(this.state.email);
    const location = { pathname: './dashboard' };
    history.push(location);
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit} className="login-form">
        <input
          onChange={this.handleChange}
          type="text"
          className="login-email-input login-input"
          value={this.state.email}
          name="email"
          placeholder="Email Address"
        />
        <input
          onChange={this.handleChange}
          type="text"
          className="login-password-input login-input"
          value={this.state.password}
          name="password"
          placeholder="Password"
        />
        <button className="login-submit-button">Submit</button>
      </form>
    );
  }
}

Login.propTypes = {
  setCurrentUser: PropTypes.func.isRequired,
  currentUser: PropTypes.object,
  history: PropTypes.object
};

export const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user))
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Login)
);
