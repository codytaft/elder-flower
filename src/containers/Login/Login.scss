.login-form {
  display: flex;
  flex-direction: column;
  align-content: center;
  width: 25rem;
  margin: auto;
}

.login-input {
  text-decoration: none;
  font-size: 32px;
  margin: 2rem 0 2rem 0;
  border-radius: 3rem;
  padding-left: 1rem;
  border: 0.1rem solid rgb(78, 45, 29);
  color: #4e2d1d;
  height: 4rem;
  background-color: seashell;
}

.login-submit-button {
  margin: 2rem auto 2rem auto;
  text-decoration: none;
  border: 0.2rem solid rgb(78, 45, 29);
  color: #4e2d1d;
  border-radius: 5rem;
  cursor: pointer;
  width: 25rem;
  height: 4rem;
  font-size: 32px;
  background-color: seashell;
}

.login-submit-button:hover {
  background-color: rgba(215, 184, 141, 1);
}
