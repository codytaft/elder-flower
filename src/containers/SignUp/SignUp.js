import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createUser, setCurrentUser } from '../../actions/userActions';

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      firstName: '',
      lastName: '',
      phoneNumber: '',
      emailAddress: '',
      emergencyName: '',
      emergencyPhone: ''
    };
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = e => {
    e.preventDefault();
    const {
      firstName,
      lastName,
      phoneNumber,
      emailAddress,
      emergencyName,
      emergencyPhone
    } = this.state;
    this.props.createUser(this.state);
  };

  testPhoneNumber = e => {
    e.preventDefault;
    const testPhoneNumber = this.state.phoneNumber;
  };

  render() {
    return (
      <section className="signup-page">
        <form className="signup-form" onSubmit={this.handleSubmit}>
          <input
            onChange={this.handleChange}
            className="signup-first-name"
            value={this.state.firstName}
            name="firstName"
            placeholder="First Name"
          />
          <input
            onChange={this.handleChange}
            className="signup-last-name"
            value={this.state.lastName}
            name="lastName"
            placeholder="Last Name"
          />
          <section className="signup-phone-section">
            <input
              onChange={this.handleChange}
              className="signup-phoneNumber"
              value={this.state.phoneNumber}
              name="phoneNumber"
              placeholder="Phone Number"
            />
            <button onClick={this.testPhoneNumber}>Test Phone</button>
          </section>

          <input
            onChange={this.handleChange}
            className="signup-email-address"
            value={this.state.emailAddress}
            name="emailAddress"
            placeHolder="Email Address"
          />
          <input
            onChange={this.handleChange}
            className="signup-emergency-name"
            value={this.state.emergencyName}
            name="emergencyName"
            placeHolder="Emergency Contact Name"
          />
          <input
            onChange={this.handleChange}
            className="signup-emergency-phone"
            value={this.state.emergencyPhone}
            name="emergencyPhone"
            placeHolder="Emergency Contact Phone"
          />
          <button className="signup-submit-btn">Submit</button>
        </form>
      </section>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  createUser: user => dispatch(createUser(user)),
  setCurrentUser: user => dispatch(setCurrentUser(user))
});

export default connect(
  null,
  mapDispatchToProps
)(SignUp);
