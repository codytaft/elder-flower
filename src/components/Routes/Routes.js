import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Login } from '../../containers/Login/Login';
import { SignUpHome } from '../../components/SignUpHome/SignUpHome';
import SignUpElder from '../../containers/SignUpElder/SignUpElder';
import SignUpCarer from '../../containers/SignUpCarer/SignUpCarer';
import Emergency from '../../containers/Emergency/Emergency';

export const Routes = () => {
  return (
    <Switch>
      <Route exact path="/" />
      <Route exact path="/home" />
      <Route exact path="/login" component={Login} />
      <Route exact path="/sign-up-home" component={SignUpHome} />
      <Route exact path="/sign-up-elder" component={SignUpElder} />
      <Route exact path="/sign-up-carer" component={SignUpCarer} />
      <Route
        exact
        path="/emergency"
        render={() => {
          return <Emergency />;
        }}
      />
    </Switch>
  );
};

export default Routes;
