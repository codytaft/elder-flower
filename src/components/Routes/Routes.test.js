import React from 'react';
import { shallow } from 'enzyme';
import { Routes } from './Routes';
import Login from '../../containers/Login/Login';
import SignUp from '../../containers/SignUp/SignUp';
import { ExecutionStepContext } from 'twilio/lib/rest/studio/v1/flow/execution/executionStep';

describe('Routes', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Routes />);
  });

  it('should match snapshot of login Route', () => {
    let loginWrapper = shallow(<Routes path="/login" component={Login} />);
    expect(loginWrapper).toMatchSnapshot();
  });
  it('should match snapshot of signup Route', () => {
    let loginWrapper = shallow(<Routes path="/signup" component={SignUp} />);
    expect(loginWrapper).toMatchSnapshot();
  });
});
