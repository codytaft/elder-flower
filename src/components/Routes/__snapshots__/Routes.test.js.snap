// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should match snapshot of login Route 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    component={undefined}
    path="/login"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/"
/>,
        <Route
          exact={true}
          path="/home"
/>,
        <Route
          exact={true}
          path="/groups"
/>,
        <Route
          exact={true}
          path="/new-group"
/>,
        <Route
          exact={true}
          path="/group-setup"
/>,
        <Route
          exact={true}
          path="/emergency"
          render={[Function]}
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/sign-up"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/login"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/home",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/groups",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/new-group",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/group-setup",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/emergency",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/sign-up",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
/>,
          <Route
            exact={true}
            path="/home"
/>,
          <Route
            exact={true}
            path="/groups"
/>,
          <Route
            exact={true}
            path="/new-group"
/>,
          <Route
            exact={true}
            path="/group-setup"
/>,
          <Route
            exact={true}
            path="/emergency"
            render={[Function]}
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/sign-up"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/login"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/groups",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/new-group",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/group-setup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/emergency",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/sign-up",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Routes should match snapshot of signup Route 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    component={undefined}
    path="/signup"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/"
/>,
        <Route
          exact={true}
          path="/home"
/>,
        <Route
          exact={true}
          path="/groups"
/>,
        <Route
          exact={true}
          path="/new-group"
/>,
        <Route
          exact={true}
          path="/group-setup"
/>,
        <Route
          exact={true}
          path="/emergency"
          render={[Function]}
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/sign-up"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/login"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/home",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/groups",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/new-group",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/group-setup",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/emergency",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/sign-up",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
/>,
          <Route
            exact={true}
            path="/home"
/>,
          <Route
            exact={true}
            path="/groups"
/>,
          <Route
            exact={true}
            path="/new-group"
/>,
          <Route
            exact={true}
            path="/group-setup"
/>,
          <Route
            exact={true}
            path="/emergency"
            render={[Function]}
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/sign-up"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/login"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/groups",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/new-group",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/group-setup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/emergency",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/sign-up",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
